File: android/src/main/kotlin/com/appdynamics/appdynamics_mobilesdk/features/request_tracking/tracker_user_data/TrackerUserDataLong.kt
/*
 * Copyright (c) 2021. AppDynamics LLC and its affiliates.
 * All rights reserved.
 *
 */

package com.appdynamics.appdynamics_mobilesdk.features.request_tracking.tracker_user_data

import androidx.annotation.NonNull
import com.appdynamics.appdynamics_mobilesdk.AppDynamicsMobileSdkPlugin
import io.flutter.plugin.common.MethodChannel
import java.lang.Exception
import java.util.*

fun AppDynamicsMobileSdkPlugin.setRequestTrackerUserDataLong(
    @NonNull result: MethodChannel.Result,
    arguments: Any?
) {
    val properties = arguments as HashMap<*, *>
    val id = properties["id"] as String

    val tracker = requestTrackers[id] ?: run {
        result.error(
            "500",
            "Agent setRequestTrackerUserDataLong() failed.",
            "Request tracker was not initialized or already reported."
        )
        return
    }

    val key = properties["key"] as? String ?: run {
        result.error(
            "500",
            "Agent setRequestTrackerUserDataLong() failed.",
            "Please provide a valid string for `key`."
        )
        return
    }

    try {
        val value = properties["value"].toString().toLong()
        tracker.withUserDataLong(key, value)
        result.success(null)
    } catch (e: Exception) {
        result.error(
            "500",
            "Agent setRequestTrackerUserDataLong() failed.",
            e.message
        )
    }
}

File: ios/Classes/Features/Request Tracking/Tracker User Data/TrackerUserDataLong.swift
import ADEUMInstrumentation
import Flutter

extension SwiftAppDynamicsAgentPlugin {
  func setRequestTrackerUserDataLong(result: @escaping FlutterResult, arguments: Any?) {
    let properties = arguments as! Dictionary<String, Any>
    let id = properties["id"] as! String
    
    guard let key = properties["key"] as? String else {
      let error = FlutterError(code: "500", message: "Agent setRequestTrackerUserDataLong() failed.", details: "Please provide a valid string for `key`.")
      result(error)
      return
    }
    
    guard let value = properties["value"] as? Int64 else {
      let error = FlutterError(code: "500", message: "Agent setRequestTrackerUserDataLong() failed.", details: "Please provide a valid long for `value`.")
      result(error)
      return
    }
    
    guard let tracker = requestTrackers[id] else {
      let error = FlutterError(code: "500", message: "Agent setRequestTrackerUserDataLong() failed.", details: "Request tracker was not initialized or already reported.")
      result(error)
      return
    }
    
    tracker.setUserDataLong(key, value: value)
    result(nil)
  }
}
